streams/delimiter_stream.ts

```diff
@@ -105,6 +105,7 @@ export class DelimiterStream extends TransformStream<Uint8Array, Uint8Array> {
             // Our chunk started with a delimiter and no previous chunks exist:
             // Enqueue an empty chunk.
             controller.enqueue(new Uint8Array());
+            chunkStart = disposition === "prefix" ? 0 : inspectIndex;
```

Comment:
**comment:** This is the fix for the second issue: We need to push our chunk start index forward unless we're going include the delimiter in the next chunk.